usb/esp_tinyusb/tusb_cdc_acm.c

```diff
@@ -298,37 +254,27 @@ esp_err_t tinyusb_cdcacm_write_flush(tinyusb_cdcacm_itf_t itf, uint32_t timeout_
     }
 
     if (!timeout_ticks) { // if no timeout - nonblocking mode
-        int res = tud_cdc_n_write_flush(itf);
-        if (!res) {
-            ESP_LOGW(TAG, "flush failed (res: %d)", res);
-            return ESP_FAIL;
-        } else {
-            if (tud_cdc_n_write_occupied(itf)) {
-                ESP_LOGW(TAG, "remained data to flush!");
-                return ESP_FAIL;
-            } else {
-                return ESP_OK;
-            }
-        }
+        // It might take some time until TinyUSB flushes the endpoint
+        // Since this call is non-blocking, we don't wait (or check)
+        // whether all data has been flushed
+        tud_cdc_n_write_flush(itf);
```

Comment:
Here, what will happen if there are more bytes remaining to be flushed !! 

Do we still need to return ESP_OK.

Suggestion:
```
    if (!timeout_ticks) { // if no timeout - nonblocking mode
        // It might take some time until TinyUSB flushes the endpoint
        // Since this call is non-blocking, we don't wait (or check)
        // whether all data has been flushed
        tud_cdc_n_write_flush(itf);

        if (tud_cdc_n_write_occupied(itf)) {
            ESP_LOGW(TAG, "remained data to flush!");
            return ESP_FAIL;
        }
    }
```

Reply:
Yes, we return ESP_OK; that is one of the main reasons for this change. **The remaining data will be flushed automatically in next packet, after this packets is sent**.

Your suggestion would return `ESP_FAIL` even though all data is going to be flushed. To be more precise, your code returning `ESP_FAIL` signals, that the multi-packet transfer is still ongoing.

Does that make sense?

Reply:
Your are right.

Different perspective that I have is:
If timeout_ticks is 0, it means that the API is non-blocking. And all data has to be flushed right away. If data is remaining to be flushed, API should throw error.

Reply:
Agreed, I will update the function so it returns `ESP_ERR_NOT_FINISHED` in this case